File: 백준/Bronze/2581. 소수/소수.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n').map(Number);

const M = input[0];
const N = input[1];

function isPrime(num) {
    if (num < 2) return false;
    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) return false;
    }
    return true;
}

let primeList = [];
for (let i = M; i <= N; i++) {
    if (isPrime(i)) {
        primeList.push(i);
    }
}

if (primeList.length > 0) {
    const primeSum = primeList.reduce((sum, val) => sum + val, 0);
    const primeMin = Math.min(...primeList);
    console.log(primeSum);
    console.log(primeMin);
} else {
    console.log(-1);
}

File: 백준/Bronze/2581. 소수/소수.py
import sys
import math

# 소수 판별 함수
def is_prime(num):
    if num <= 1:
        return False
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            return False
    return True

# 입력 받기
M = int(sys.stdin.readline().strip())
N = int(sys.stdin.readline().strip())

# M 이상 N 이하의 소수 찾기
prime_list = [num for num in range(M, N + 1) if is_prime(num)]

if prime_list:
    prime_sum = sum(prime_list)
    prime_min = min(prime_list)
    print(prime_sum)
    print(prime_min)
else:
    print(-1)
